### PR TITLE
Fix a CompletionFailure and a follow-up NPE

### PR DESCRIPTION
**Pull request by wmdietlGC**

### CODE REVIEW

File: javacutil/src/org/checkerframework/javacutil/ElementUtils.java [hunk 1]
```diff
@@ -360,14 +360,25 @@ public static boolean hasReceiver(Element element) {
             // For each direct supertype of the current type element, if it
             // hasn't already been visited, push it onto the stack and
             // add it to our superelems set.
-            TypeMirror supertypecls = current.getSuperclass();
-            if (supertypecls.getKind() != TypeKind.NONE) {
+            TypeMirror supertypecls;
+            try {
+                supertypecls = current.getSuperclass();
```
Comment (smillst) [inline javacutil/src/org/checkerframework/javacutil/ElementUtils.java]: There are 18 other calls to getSuperClass in the Checker Framework.  Could all of them also issue a CompletionFailure or just this location?

File: framework/src/org/checkerframework/framework/source/SourceChecker.java [hunk 2]
```diff
@@ -939,6 +939,13 @@ public void typeProcess(TypeElement e, TreePath p) {
         } else {
             previousErrorCompilationUnit = null;
         }
+        if (visitor == null) {
+            // typeProcessingStart invokes initChecker, which should
+            // have set the visitor. If the field is still null, an
+            // exception occured during initialization, which was already
+            // logged there. Don't also cause a NPE here.
+            return;
```
Comment (smillst) [inline framework/src/org/checkerframework/framework/source/SourceChecker.java]: Is this change unrelated to the other one?

Comment (wmdietl) [inline framework/src/org/checkerframework/framework/source/SourceChecker.java]: The CompletionFailure from the other change leaves the visitor null, which then causes a related NPE here.

File: javacutil/src/org/checkerframework/javacutil/ElementUtils.java (hunk 1 above)
Comment (wmdietl) [inline javacutil/src/org/checkerframework/javacutil/ElementUtils.java]: I don't know.

**Pull request merged**